source/bem_problem.cc

```diff
@@ -566,23 +602,10 @@ void BEMProblem<dim>::assemble_system()
   dirichlet_matrix = 0;
 
 
-  std::vector<Quadrature<dim-1> > sing_quadratures;
+  std::vector<Quadrature<dim-1> > sing_quadratures(fe->dofs_per_cell);
   for (unsigned int i=0; i<fe->dofs_per_cell; ++i)
     {
-      if (fe->degree > 1)
-        {
-          sing_quadratures.push_back(QIterated<dim-1>(QGauss<1> (singular_quadrature_order),fe->degree));
-        }
-      else
-        {
-          sing_quadratures.push_back
-          (QTelles<dim-1>(singular_quadrature_order,
-                          fe->get_unit_support_points()[i]));
-          //
-          // Usage of alternative singular quadrature formula
-          // (QGaussOneOverR<dim-1>(singular_quadrature_order,
-          //                 fe->get_unit_support_points()[i],true));
-        }
+      sing_quadratures[i] = get_singular_quadrature(i);
```

Comment:
Why do you create a new vector here? Since you create it as a static vector in the `get_singular_quadrature` function, there is no need to copy it back here.

Reply:
just call `get_singular_quadrature(i)` when you need it, instead of `sing_quadratures[i] `